medicines/doc-index-updater/src/document_manager/mod.rs

```diff
@@ -54,6 +54,7 @@ where
                     },
                 )
                 .await?;
+            tracing::error!("Failed to dispatch to queue. Check API keys");
```

Comment:
can you put this above the state_manager call? a failure in state_manager will prevent us from hitting this line. You could also use `Err(e)` and print the true error message of `e`.